### PR TITLE
snap: add `snap download --assertion model/16/canonical/pc-amd64` support

### PR DESCRIPTION
**Pull request by mvo5**
This is a strawman branch. I am a bit unsure what the best CLI experience here is. `snap download-assertion`, `snap download --assertion`, `snap known --remote`, ... ? The idea behind it is to provide an easy way for people to download assertions, e.g. the model assertion.

### CODE REVIEW

**Comment by chipaca** [comment]
As for myself, I'd love to start exploring the "remote" idea. But I'm being selfish.

File: cmd/snap/cmd_download.go [hunk 1]
```diff
@@ -87,6 +112,10 @@ func (x *cmdDownload) Execute(args []string) error {
 		return ErrExtraArgs
 	}
 
+	if x.Assertion {
```
Comment (chipaca) [inline cmd/snap/cmd_download.go]: check for `x.Revision` and error out if given?

**Comment by chipaca** [comment]
This LGTM as is, fwiw (except I'd check for `--revision` given together with `--asertion`)

**Comment by zyga** [comment]
Apart from what chipaca noted about checking the revision of the snap in the the assertion I'd like to know how we can tag potentially unstable CLI extensions. Pehaps we could use an environment variable like `SNAP_EXPERIMENTAL` to gate such features. Without the variable being set and containing the token "cli" we might just not show those options. The people that know about the experimental feature can use it but others won't accidentally learn something that might change overnight.

**Comment by niemeyer** [comment]
@mvogt I'm closing this one due to the above observations, and adding an entry to our feature freeze.

I'm not sure if that's proper for FF or not as I don't have context. Let's catch up on it tomorrow.

**Pull request closed without merging**